browser/brave_wallet/brave_wallet_prefs_unittest.cc

```diff
@@ -274,4 +276,79 @@ TEST_F(BraveWalletPrefsUnitTest, MigrateShowTestNetworksToggle) {
   EXPECT_FALSE(GetPrefs()->HasPrefPath(kShowWalletTestNetworksDeprecated));
 }
 
+TEST_F(BraveWalletPrefsUnitTest, MigrateShowChainIdNetworkInfo) {
+  GetPrefs()->SetBoolean(kBraveWalletUserAssetEthContractAddressMigrated, true);
+  GetPrefs()->SetBoolean(kBraveWalletUserAssetsAddPreloadingNetworksMigrated,
+                         true);
+  GetPrefs()->SetBoolean(kBraveWalletUserAssetsAddIsNFTMigrated, true);
+  GetPrefs()->SetBoolean(kBraveWalletEthereumTransactionsCoinTypeMigrated,
+                         true);
+  GetPrefs()->SetBoolean(kBraveWalletDeprecateEthereumTestNetworksMigrated,
+                         true);
+
+  EXPECT_FALSE(
+      GetPrefs()->HasPrefPath(kBraveWalletTransactionsChainIdMigrated));
+  EXPECT_FALSE(GetPrefs()->GetBoolean(kBraveWalletTransactionsChainIdMigrated));
+
+  const char ethTxId[] = "b1e8dda1";
+  const auto ethPath = base::StringPrintf("%s.%s", "mainnet", ethTxId);
+  base::Value::Dict tx1;
+  tx1.Set("id", ethTxId);
+
+  const char solTxId[] = "887e878f";
+  const auto solPath = base::StringPrintf("%s.%s", "devnet", solTxId);
+  base::Value::Dict tx2;
+  tx2.Set("id", solTxId);
+
+  const char filTxId[] = "197ea1e5";
+  const auto filPath = base::StringPrintf("%s.%s", "testnet", filTxId);
+  base::Value::Dict tx3;
+  tx3.Set("id", filTxId);
+
+  {
+    ScopedDictPrefUpdate update(GetPrefs(), kBraveWalletEthereumTransactions);
+    update->SetByDottedPath(ethPath, tx1.Clone());
+  }
+  {
+    ScopedDictPrefUpdate update(GetPrefs(), kBraveWalletSolanaTransactions);
+    update->SetByDottedPath(solPath, tx2.Clone());
+  }
+  {
+    ScopedDictPrefUpdate update(GetPrefs(), kBraveWalletFileCoinTransactions);
+    update->SetByDottedPath(filPath, tx3.Clone());
+  }
+
+  MigrateObsoleteProfilePrefs(GetPrefs());
+
+  EXPECT_TRUE(GetPrefs()->HasPrefPath(kBraveWalletTransactionsChainIdMigrated));
+  EXPECT_TRUE(GetPrefs()->GetBoolean(kBraveWalletTransactionsChainIdMigrated));
+
+  const std::string* ethChainId =
+      GetPrefs()
+          ->GetDict(kBraveWalletEthereumTransactions)
+          .FindStringByDottedPath(
+              base::StringPrintf("%s.%s", ethPath.c_str(), "chain_id"));
```

Comment:
base::StrCat seems better to me.

Reply:
You can use base::JoinString to compose ethereum.ethPath.chain_id if we get rid of kBraveWalletEthereumTransactions.

Reply:
changed it to base::JoinString, thanks